mm2src/coins/utxo/utxo_common.rs

```diff
@@ -2184,6 +2196,69 @@ pub async fn search_for_swap_tx_spend_other<T: AsRef<UtxoCoinFields> + SwapOps>(
     .await
 }
 
+pub async fn get_taker_watcher_reward<T: UtxoCommonOps + SwapOps + MarketCoinOps>(
+    coin: &T,
+    other_coin: &MmCoinEnum,
+    coin_amount: Option<BigDecimal>,
+    other_coin_amount: Option<BigDecimal>,
+    reward_amount: Option<BigDecimal>,
+) -> Result<Option<WatcherReward>, MmError<WatcherRewardError>> {
```

Comment:
To make the `coin_amount` and `other_coin_amount` non-optional, I can implement a method `watcher_get_taker_watcher_reward` for the use of watchers, but the problem is that it will be very slightly different than `get_taker_watcher_reward`, and this duplication will be repeated for all coin types. Reducing lots of duplication with optional parameters seems to me to be the better solution. For the `get_maker_watcher_reward` methods the coin amount were not used at all, so I removed the parameters.

To make the `reward_amount` non-optional, I will need to implement again two different methods, both for the taker reward and the maker reward, and these methods will be again slightly different than each other. Also, I will need to use if else statements on `taker_swap` / `maker_swap` / `swap_watcher` to determine which of these methods will be called. I think it is better as it is currently.

I made the `get_taker_watcher_reward` method return `WatcherReward` directly instead of an Option since it always returned `Some`. For the `get_maker_watcher_reward` method, however, `ETH` will return a `WatcherReward` while all remaining coins will return `None`, so returning an Option in this case is necessary.

Reply:
This fine by me. I was just trying to follow the separation of concern design principle where if we assumed that the watcher, taker and maker are separate entities they should use different functions with different responsibilities to allow each function to grow on it's own in the future. Generally speaking, when we pass an option or a bool to a function, this can be an indication that a condition or a match that can be handled elsewhere is moved inside the function and the logic inside the function is coupled with other logic outside the function.
We can leave it as it is for now in this PR, although I am pretty sure as the watchers start to support more protocols in the future we will have to break these functions down.